.github/workflows/test_on_push.yml

```diff
@@ -0,0 +1,105 @@
+name: PyBaMM
+
+on:
+  push:
+
+  # everyday at 3 am UTC
+  schedule:
+    - cron:  '* 3 * * *'
+
+jobs:
+  
+  style:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check style
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+
+    - name: Install package
+      run: |
+        python -m pip install --upgrade pip wheel setuptools
+        pip install -e .[dev]
+
+    - name: Check style
+      run: python -m flake8
+
+  build:
+    needs: style
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.6, 3.7]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    
+    - name: Install Linux system dependencies
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        sudo apt install gfortran gcc libopenblas-dev liblapack-dev graphviz
+        sudo apt install python${{ matrix.python-version }}.dev
+    
+    - name: Install MacOS system dependencies
+      if: matrix.os == 'macos-latest'
+      run: |
+        brew update
+        brew install graphviz
+        brew install openblas suitesparse
+        
+    - name: Install Windows system dependencies
+      if: matrix.os == 'windows-latest'
+      run: choco install graphviz
+    
+    - name: Install standard python dependencies
+      run: |
+        python -m pip install --upgrade pip wheel setuptools
+        pip install -e .
+
+    - name: Install SUNDIALS and SuiteSparse
+      if: matrix.os != 'windows-latest'
+      run: |
+        pip install wget
+        python scripts/setup_KLU_module_build.py
+        pybamm_install_odes --install-sundials
```

Comment:
`pybamm_install_odes` is an entry point defined in the setup.py, so I'm surprised the command is found before pybamm is installed!

Reply:
`setup_KLU_module_build.py` already installs the sundials in `$HOME/.local/`, so the `pybamm_install_odes` command could be replaced by
```
export SUNDIALS_INST= ${{ HOME }}/.local
pip install scikits.odes
```